src/registry/registry.hpp

```diff
@@ -24,18 +24,23 @@ class ResourceSubtype {
     const google::protobuf::ServiceDescriptor* service_descriptor;
     virtual std::shared_ptr<ResourceServerBase> create_resource_server(
         std::shared_ptr<SubtypeService> svc);
-    std::function<std::shared_ptr<ResourceBase>(std::string, std::shared_ptr<grpc::Channel>)>
-        create_rpc_client;
+    virtual std::shared_ptr<ResourceBase> create_rpc_client(std::string name,
+                                                            std::shared_ptr<grpc::Channel> channel);
 
     ResourceSubtype(const google::protobuf::ServiceDescriptor* service_descriptor)
         : service_descriptor(service_descriptor){};
 
    private:
 };
 
+// default_validator is the default validator for all models if no validator is
+// provided in construction. No dependencies are returned.
+std::vector<std::string> default_validator(Resource cfg);
```

Comment:
It looks to me like this is only used within `registry.cpp`. we should probably not expose this in the header unless we have to. and if we do need to expose it, it should be marked `const` and prefixed with `k_` (k for const). 

Alternatively, we can keep it in the header but make it a `private` `static` `const` member of `ModelRegistration`, which seems to be the only type that uses it?

Reply:
Ah great idea. Made it a `private static const` member of `ModelRegistration`.